### PR TITLE
[Fogbugz # 1352093] Fixing bad copy texture when DRS is on with ray tracing.

### PR DESCRIPTION
**Pull request by kecho**
### Purpose of this PR
Fogbugz: https://fogbugz.unity3d.com/f/cases/1352093/
Copy was failing because the mip texture size computed was off by 1. Made it conservative by using a ceiling instead of casting down a float to an int.

---
### Testing status
* Tested repro case from fogbugz. Windows with hardware DRS on dx11.
* QA should test this with many DRS changes on hardware mode.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs:645]: IIRC there is a disparity of rounding behaviour between HW and SW, have you checked both with a collection of resolution? (odd resolutions will be the problem I think). 

If you didnt I guess that's a ping for QA

**Review by sebastienlagarde — COMMENTED** [review]
please target master

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs
Comment (kecho) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Utility/HDUtils.cs:645]: It should work since that computation is what writes on acualWidth and actualHeight.
Will indeed ask QA to test for this.

**Comment by kecho** [comment]
> please target master

Retargetting and rerunning yamato.
Adding QA as well

**Review by FrancescoC-unity — APPROVED** [review]
LGTM, pending the QA check I mentioned in the other comment, but code looks good

**Review by sebastienlagarde — APPROVED** [review]

**Review by remi-chapelain — COMMENTED** [review]
Found a bunch of DRS issues ([1354180](https://fogbugz.unity3d.com/f/cases/1354180/), [1354181](https://fogbugz.unity3d.com/f/cases/1354181/), [1354184](https://fogbugz.unity3d.com/f/cases/1354184/)) while testing, but all of theses seems to repro on current master so the PR didn't created that. 

However, since hardware mode is kind of broken right now cf [1354184](https://fogbugz.unity3d.com/f/cases/1354184/), I'm not sure I should approve even though I wasn't able to repro the console "bad copy texture" testing on all the filters (including DLSS) and a lot of different resolutions (classic and odd).

**Comment by FrancescoC-unity** [comment]
@remi-chapelain  I closed 1354180 as Lanczos is about to get removed altogether :-) 

I wonder if 1354184 is the one that @kecho  fixed in another open PR (https://github.com/Unity-Technologies/Graphics/pull/5231)

**Comment by remi-chapelain** [comment]
I'll have a quick look to 1354184 with the other PR, to be sure !

**Review by remi-chapelain — APPROVED** [review]
Okay, so indeed, I tried this small change on top of [this PR](https://github.com/Unity-Technologies/Graphics/pull/5231) to be able to test hardware mode. ✔️ 

And I confirm that [this PR](https://github.com/Unity-Technologies/Graphics/pull/5231) indeed fixes the [1354184](https://fogbugz.unity3d.com/f/cases/1354184/). I resolved it.

**Pull request merged**
